tools/release/refresh-packages.js

```diff
@@ -0,0 +1,175 @@
+/* eslint-disable no-console */
+/**
+ * External dependencies
+ */
+const fs = require( 'fs' );
+const spawn = require( 'cross-spawn' );
+const { zip, uniq, identity, groupBy } = require( 'lodash' );
+
+/**
+ * Constants
+ */
+const WORDPRESS_PACKAGES_PREFIX = '@wordpress/';
+const rootDir = __dirname + '/../..';
+const { getArgFromCLI } = require( `${ rootDir }/node_modules/@wordpress/scripts/utils` );
+const distTag = getArgFromCLI( '--dist-tag' ) || 'latest';
+
+/**
+ * The main function of this task.
+ *
+ * It installs any missing WordPress packages, and updates the
+ * mismatched dependencies versions, e.g. it would detect that Gutenberg
+ * updated react from 16.0.4 to 17.0.2 and install the latter.
+ */
+function refreshDependencies() {
+	const initialPackageJSON = readJSONFile( `${ rootDir }/package.json` );
+
+	// Install any missing WordPress packages:
+	const missingWordPressPackages = getMissingWordPressPackages();
+	if ( missingWordPressPackages.length ) {
+		console.log( "The following @wordpress dependencies are missing: " );
+		console.log( missingWordPressPackages );
+		console.log( "Installing via npm..." );
+		installPackages( missingWordPressPackages.map( name => [name, distTag] ) );
+	}
+
+	// Update any outdated non-WordPress packages:
+	const versionMismatches = getMismatchedNonWordPressDependencies();
+	if ( versionMismatches.length ) {
+		console.log( "The following dependencies are outdated: " );
+		console.log( versionMismatches );
+		console.log( "Updating via npm..." );
+		const requiredPackages = versionMismatches.map( ( { name, required } ) => [name, required] );
+		installPackages( requiredPackages );
+	}
+
+	const finalPackageJSON = readJSONFile( `${ rootDir }/package.json` );
+	outputPackageDiffReport(
+		getPackageVersionDiff( initialPackageJSON, finalPackageJSON ),
+	);
+	process.exit( 0 );
+}
+
+refreshDependencies();
```

Comment:
Maybe this call should be moved to the end of the file because it's easy to miss otherwise.